src/main/scala/edu/berkeley/cs/rise/opaque/strategies.scala

```diff
@@ -117,6 +117,12 @@ object OpaqueOperators extends Strategy with JoinSelectionHelper {
             foreignProjSchema.map(_.toAttribute),
             EncryptedSortExec(sortOrder, false, partitioned)
           )
+        case FullOuter =>
+          EncryptedAddDummyRowExec(
+            foreignProjSchema.map(_.toAttribute),
+            EncryptedSortExec(sortOrder, false, partitioned),
+            primaryProjSchema.map(_.toAttribute)
```

Comment:
Is it possible to achieve this additional functionality without modifying `EncryptedAddDummyRowExec`? I would try having two calls to `EncryptedAddDummyRowExec` in this `FullOuter` case.

Reply:
Yeah, I wasn't sure about doing it this way either, so I'll try the other approach instead.